ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -665,39 +665,50 @@ def test016_validate_order_test_unit_test_plan(self):
         self.order_page.set_test_unit(test_unit='r')
         self.order_page.save(save_btn='order:save_btn')
         
-    # will continue wih us
     def test017_validate_order_test_unit_test_plan_edit_mode(self):
         """
         New: orders Test plan /test unit validation in edit mode
 
         LIMS-4826
         """
-        self.base_selenium.LOGGER.info(
-            ' Running test case to check that at least test unit or test plan is mandatory in order')
-
-        # validate in edit mode, go to order over view
-        self.order_page.get_orders_page()
-        self.order_page.get_random_order()
-        order_url = self.base_selenium.get_url()
-        self.base_selenium.LOGGER.info(' + order_url : {}'.format(order_url))
+        self.base_selenium.LOGGER.info(' Running test case to check that at least test unit or test plan is mandatory in order')
+
+        # Get random order
+        order_request = self.orders_api.get_all_orders(limit=20).json()
+        self.assertEqual(order_request['status'], 1)
+        orders_records = order_request['orders']
+        self.assertNotEqual(len(orders_records), 0)
+        random_order_index = self.generate_random_number(lower=0, upper=len(orders_records) - 1)
+        selected_order_record = orders_records[random_order_index]
```

Comment:
you can use 
`selected_order_record = random .choice(order_request['orders']) 
`directly

Reply:
Done